### DISCUSSION TITLE
Example: Adding a GitHub repo which uses fixed tags and contains multiple apps scattered across releases (`FiorenMas/Revanced-And-Revanced-Extended-Non-Root`)

### DISCUSSION THREAD

**Discussion by ImranR98** [ImranR98/Obtainium#1494] [Q&A] [answered]
From https://github.com/ImranR98/Obtainium/issues/1492#issue-2202314273:

> https://github.com/FiorenMas/Revanced-And-Revanced-Extended-Non-Root/releases
>
> that release contains all apk that i wanna add. i really noob to this stuff. what tooggle should i set to on if i wanna to make sure all apk i added from those repo got updated?
>
> thanks

**Comment by ImranR98** [comment, marked as answer]
😤 this is one of those repos that makes automatic scraping very difficult:
- Re-uses fixed-name tags instead of [using tags in the intended way](https://git-scm.com/book/en/v2/Git-Basics-Tagging) (creating a new tag per release).
    - This also means older versions of apps are gone.
- Includes multiple apps in the same repo.
- Bundles some apps in the same release but other apps in separate releases (the 'latest' marker is also meaningless because of this).

It should still be possible to add it though. Let's say you want the patched Instagram app. Use the following settings (anything not listed can be left as default):
- App Source URL: `https://github.com/FiorenMas/Revanced-And-Revanced-Extended-Non-Root`
- Filter Release Titles by Regular Expression: `Revanced`
    - This tells it to ignore any release that doesn't have "Revanced" in the title (leaving us with [this](https://github.com/FiorenMas/Revanced-And-Revanced-Extended-Non-Root/releases/tag/all) single release)
- Use latest asset upload as release date: ON
    - This tells it not to pay attention to release dates because in our case, the repo owner is re-using old releases instead of making new ones. The release we want has the date "Jun 1, 2023" and this will never change, making it useless.
    - Instead, Obtainium will pay attention to the date on which APKs are actually uploaded - this does change whenever the repo owner uploads an updated APK, so we can use it.
- Use release date as version string (pseudo-version): ON
    - Usually, a release's tag is taken as the app version. Since this repo uses fixed tags that never change, we use release dates as versions instead (this setting works together with the previous one - where the previous setting corrected the release date, this setting tells Obtainium to actually make use of that date for versioning).
- Filter APKs by Regular Expression: `instagram`
    - This tells it to ignore all APKs it finds in the release, except for ones that match.
    - If the filter matches multiple APKs, you will be given the choice to manually pick one.
    - Manually picking APKs is inconvenient, error prone, and disables background updates.
    - The string provided above will leave you with 2 remaining APKs - a beta and non-beta. If you don't want to have to manually pick between these every time, you can use a more complex filter:
        - `instagram-((?!beta).)+\.apk` for non-beta
        - `instagram-.+beta` for beta

If you need a different app, the settings are the same except you'd replace `instagram` with something else. Each app needs to be added separately.

For MicroG, use the same settings as above but with these changes:
- Filter Release Titles by Regular Expression: `MicroG`
- Filter APKs by Regular Expression: _empty_

Note that:
- You won't see the real version number for any of these apps, you'll see a long number instead - this is a 'pseudo-version' that has to be used because of the issues mentioned at the top. Things will still work in mostly the same way (see the last part of [this Wiki section](https://github.com/ImranR98/Obtainium/wiki/#version-detection) for a couple of behaviour differences to keep in mind when the "version" being used is not normal).
- If you already had the app installed when you added it to Obtainium, you will need to tap the "Mark Updated" (check mark) button the first time after it is added.

**Comment by mas-mas-30tahun** [comment]
what a brief explananation. very much thanks
